### DISCUSSION TITLE
How to use GLPI Agent Portable?

### DISCUSSION THREAD

**Discussion by maestroarnaldo** [glpi-project/glpi-agent#273] [Q&A] [answered]
I can't find a documentation about how to use GLPI Agent Portable

I want to run an inventory in new computers/notebooks, just using a portable agent, without to install agent on computer.
There are a lot of .bat files, but no explanation in how to use, and what to use.

I just want a "how-to" use Portable Agent

**Comment by g-bougard** [comment]
Hi @maestroarnaldo 

thank you for your suggestion, I'll try to add such howto in online documentation.

**Comment by fernandobayit** [comment]
+1

**Comment by angimo9** [comment]
Edit the glpi-inventory file in perl\bin with any text editor
On line 70, edit the path e.g. 'c:\temp'.
Execute glpi-inventoy.bat and it will create the file to import in your glpi-server in c:\temp
The filename is some HostnameDate.xml

**Comment by maestroarnaldo** [comment]
Hi!

Thanks! It created the file as waited.
But is there some way to run the agent by sending it directly to server?

Thanks!

**Comment by g-bougard** [comment]
Hi @maestroarnaldo 
just use `glpi-agent.bat` script using `--server` option or just create a configuration file under `etc\conf.d` to include "server = http://youglpiserver". You can read [glpi-agent man page](https://glpi-agent.readthedocs.io/en/latest/man/glpi-agent.html) for informations about glpi-agent script options and the [glpi-agent configuration page](https://glpi-agent.readthedocs.io/en/latest/configuration.html) for doc about all possible parameters.
With glpi-agent portable the `glpi-agent.bat` script is setup to use file as configuration backend.

If glpi-agent is run from a network shared disk or from a usb disk, you must use `vardir` option to setup a local folder where all local agent informations will be stored.

About @angimo9 solution, don't use it to store local inventory. Better use `local` option of `glpi-agent` script or just redirect output of `glpi-inventory` to the file you want to create.

**Comment by maestroarnaldo** [comment, marked as answer]
Guys, I solved my problem.

I just use this command:
glpi-agent.bat --server http://myserver/glpi/front/inventory.php --vardir=%temp%

Note: GLPI Agent 1.2 didn't work with GLPI 10.0.5. I downloaded and ran GLPI 1.4, and worked fine.
